spacy/kb_new/kb_interface.py

```diff
@@ -0,0 +1,204 @@
+"""Interface for KnowledgeBase for entity or concept linking."""
+import abc
+from typing import List, Any, Iterable, Iterator, Collection
```

Comment:
As a general comment, I don't think we typically add docstrings to the module file? We can just document each (or the one single) class in it. And then on the first line we typically put the imports `from typing`.